barretenberg/cpp/src/barretenberg/dsl/acir_format/ivc_recursion_constraint.test.cpp

```diff
@@ -214,64 +182,110 @@ TEST_F(IvcRecursionConstraintTest, AccumulateFour)
+        num_app_public_inputs = app_circuit.public_inputs.size();
 
-        // Save the single entry in the verification queue at this point
-        alternative_verification_queue_entry = ivc.verification_queue[0];
+        // Construct and accumulate kernel consisting only of the kernel completion logic
+        AcirProgram program = construct_mock_kernel_program(ivc.verification_queue, { num_app_public_inputs });
+        Builder kernel = acir_format::create_kernel_circuit(program.constraints, ivc, program.witness);
+        ivc.accumulate(kernel);
+        expected_kernel_vk = ivc.verification_queue.back().honk_verification_key;
+    }
+
+    // Now, construct the kernel VK by mocking the post app accumulation state of the IVC
+    std::shared_ptr<ClientIVC::VerificationKey> kernel_vk;
+    {
+        ClientIVC ivc;
+        ivc.trace_settings = trace_settings;
+
+        acir_format::mock_ivc_oink_accumulation(ivc, num_app_public_inputs - bb::PAIRING_POINT_ACCUMULATOR_SIZE);
 
-        // Construct and accumulate kernel_0
-        size_t num_pub_inputs_app = app_circuit.public_inputs.size();
-        AcirProgram program_0 = construct_mock_kernel_program(ivc.verification_queue, { num_pub_inputs_app });
-        Builder kernel_0 = acir_format::create_kernel_circuit(program_0.constraints, ivc, program_0.witness);
-        ivc.accumulate(kernel_0);
+        // Construct kernel consisting only of the kernel completion logic
+        AcirProgram program = construct_mock_kernel_program(ivc.verification_queue, { num_app_public_inputs });
+        Builder kernel = acir_format::create_kernel_circuit(program.constraints, ivc);
+        // WORKTODO: this would normally happen in accumulate()
+        kernel.add_pairing_point_accumulator(stdlib::recursion::init_default_agg_obj_indices<Builder>(kernel));
 
-        EXPECT_TRUE(ivc.prove_and_verify());
+        auto proving_key = std::make_shared<DeciderProvingKey_<MegaFlavor>>(kernel, trace_settings);
+        MegaProver prover(proving_key);
+        kernel_vk = std::make_shared<ClientIVC::VerificationKey>(prover.proving_key->proving_key);
     }
 
-    // Repeat a similar IVC but use the alternative queue entry just created to provide different (but independently
-    // valid) witnesses during constraint system construction VS recursive verifier construction.
+    // PCS verification keys will not match so set to null before comparing
+    kernel_vk->pcs_verification_key = nullptr;
+    expected_kernel_vk->pcs_verification_key = nullptr;
 
-    ClientIVC ivc;
-    ivc.trace_settings.structure = TraceStructure::SMALL_TEST;
+    EXPECT_EQ(*kernel_vk.get(), *expected_kernel_vk.get());
+}
 
-    // construct and accumulate a mock app circuit with a single unique public input
-    Builder app_circuit = construct_mock_app_circuit(ivc);
-    ivc.accumulate(app_circuit);
+// Test generation of "init" kernel VK via dummy IVC data
+TEST_F(IvcRecursionConstraintTest, GenerateVKFromConstraints)
```

Comment:
what's missing from this test?

Reply:
not sure what you mean - I wouldn't say anything is missing, its just that right now I'm only supporting the mocking of the IVC state after accumulation of the first app. All that's needed I suppose is functionality to mock fold proofs which shouldn't be hard